modules/document/bodycomponent.go

```diff
@@ -49,60 +50,128 @@ func createBodyComponent(component *openapi3.Schema, doc *document.Document) {
+			fieldData := arrayItemsSchema.Properties[fieldName].Value
 
-		cell := row.AddCell()
-		paragraph := cell.AddParagraph().AddRun()
-		paragraph.Properties().SetBold(true)
-		paragraph.AddText(fieldName)
+			isRequired := false
+			if utils.IsDataExistInArray(arrayItemsSchema.Required, fieldName) == true {
+				isRequired = true
+			}
 
-		cell = row.AddCell()
-		cell.AddParagraph().AddRun().AddText(fieldData.Type)
+			row := table.AddRow()
 
-		cell = row.AddCell()
-		cell.AddParagraph().AddRun().AddText(fieldData.Description)
+			cell := row.AddCell()
+			paragraph := cell.AddParagraph().AddRun()
+			paragraph.Properties().SetBold(true)
+			paragraph.AddText(fieldName)
 
-		cell = row.AddCell()
-		cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Required: %s", strconv.FormatBool(isRequired)))
+			cell = row.AddCell()
+			cell.AddParagraph().AddRun().AddText(fieldData.Type)
 
-		if fieldData.Default != nil {
-			cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Default: %v", fieldData.Default))
-		}
+			cell = row.AddCell()
+			cell.AddParagraph().AddRun().AddText(fieldData.Description)
+
+			cell = row.AddCell()
+			cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Required: %s", strconv.FormatBool(isRequired)))
 
-		if len(fieldData.Enum) >= 1 {
-			listEnum := ""
-			for i, datum := range fieldData.Enum {
-				if i != 0 {
-					listEnum += ", "
+			if fieldData.Default != nil {
+				cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Default: %v", fieldData.Default))
+			}
+
+			if len(fieldData.Enum) >= 1 {
+				listEnum := ""
+				for i, datum := range fieldData.Enum {
+					if i != 0 {
+						listEnum += ", "
+					}
+					listEnum += datum.(string)
 				}
-				listEnum += datum.(string)
+
+				cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Enum: %s", listEnum))
 			}
 
-			cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Enum: %s", listEnum))
-		}
+			if fieldData.Example != nil {
+				cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Example: %s", fieldData.Example.(string)))
+			}
 
-		if fieldData.Example != nil {
-			cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Example: %s", fieldData.Example.(string)))
-		}
+			if fieldData.Pattern != "" {
+				cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Pattern: %s", fieldData.Pattern))
+			}
 
-		if fieldData.Pattern != "" {
-			cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Pattern: %s", fieldData.Pattern))
-		}
+			if fieldData.MinLength != 0 {
+				cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Min. Length: %s", strconv.FormatUint(fieldData.MinLength, 10)))
+			}
 
-		if fieldData.MinLength != 0 {
-			cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Min. Length: %s", strconv.FormatUint(fieldData.MinLength, 10)))
+			if fieldData.MaxLength != nil {
+				cell.AddParagraph().AddRun().AddText(fmt.Sprintf("Max. Length: %s", strconv.FormatUint(*fieldData.MaxLength, 10)))
+			}
 		}
+	} else {
+		// Process the schema directly
+		for fieldName := range component.Properties {
```

Comment:
If the code below is the same, encapsulate this in method `processSchemaProperties` and reuse for both cases - array, and component.Properties.